code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -370,21 +376,46 @@
 		src.antigrab_counter = 0
 	if(keys & KEY_RUN)
 		if(!src.floorrunning && isfeathertile(src.loc))
-			if(istype(src.loc, /turf/simulated/floor/feather))
+			if (length(src.grabbed_by))
+				for(var/obj/item/grab/g in src.grabbed_by)
+					if (!(g.state == GRAB_PASSIVE || g.state == GRAB_PIN)) // in the rare case you do pin a flockdrone
```

Comment:
so floorrunning can break a pin but not an aggressive grab? Feels a little odd

Reply:
I was trying to think of it visually, if somehow you were to be able to pin a flockdrone on the floor it seems like they could escape right through the floor, but idk. If it's better to not I'll change it.